### PR TITLE
[colmap] Update COLMAP port to 3.8

### PR DESCRIPTION
**Pull request by ahojnnes**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/colmap/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "colmap",
-  "version-date": "2022-03-14",
-  "port-version": 1,
+  "version-date": "2023-02-03",
```
Comment (JonLiu1993) [inline ports/colmap/vcpkg.json:3]:
```suggestion
  "version": "3.8",
```

File: ports/colmap/portfile.cmake [hunk 2]
```diff
@@ -1,13 +1,11 @@
-set(COLMAP_REF "29a1e3642a3b00734a52b21e597ea4d576485fe6") # 3.7 fix
+set(COLMAP_REF "45edd3dfde0d945d305de3f299f3c36ae098620c") # 3.8
```
Comment (JonLiu1993) [inline ports/colmap/portfile.cmake:1]:
```suggestion
```

File: ports/colmap/portfile.cmake [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO colmap/colmap
     REF ${COLMAP_REF}
```
Comment (JonLiu1993) [inline ports/colmap/portfile.cmake:6]:
```suggestion
    REF "${VERSION}"
```

File: ports/colmap/portfile.cmake (hunk 2 above)
Comment (JonLiu1993) [reply ports/colmap/portfile.cmake:1]: @ahojnnes, thanks for the PR, we can use "${VERSION}" to get the REF of the version from vcpkg.json.

File: ports/colmap/vcpkg.json (hunk 1 above)
Comment (JonLiu1993) [reply ports/colmap/vcpkg.json:3]: If there is a stable release, we should use the version number here instead of the date.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/84733/artifacts?artifactName=failure%20logs%20for%20x64-linux&api-version=7.0&%24format=zip). Please ping me if you need any help.

File: ports/colmap/portfile.cmake [hunk 4]
```diff
@@ -1,13 +1,9 @@
-set(COLMAP_REF "29a1e3642a3b00734a52b21e597ea4d576485fe6") # 3.7 fix
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO colmap/colmap
-    REF ${COLMAP_REF}
-    SHA512 c22511592dadd1fce51baeaa5ab3ca48b0df5f1c02f9e2a97593ea1b01c5aea0e1054063a5665e2653f2c7b1b7525ce4c62ae35fb4197df614112861045b76fd
+    REF ${VERSION}
```
Comment (JonLiu1993) [inline ports/colmap/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Please get failure logs here:
````
CMake Error at cmake/FindFreeImage.cmake:91 (message):
  Could not find FreeImage
Call Stack (most recent call first):
  /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:843 (_find_package)
  CMakeLists.txt:102 (find_package)
````

File: ports/colmap/portfile.cmake [hunk 5]
```diff
@@ -1,13 +1,9 @@
...
+    SHA512 2a5d051592182264deceec861001fc4c2913ca4255589d6eac30b9e8619bf93b3143797d9f9eb7658150d30b2ae3bcc5297602e1b386a6267778ed342c1edb67
     HEAD_REF dev
-    PATCHES
-        fix-dependencies.patch
```
Comment (dg0yt) [inline ports/colmap/portfile.cmake:10]: You cannot drop this patch. We need exported config from freeimage in order to collect private link libraries.

File: ports/colmap/fix-dependencies.patch [hunk 6]
```diff
@@ -1,18 +1,21 @@
...
-+find_package(freeimage CONFIG REQUIRED)
-+set(FREEIMAGE_LIBRARIES freeimage::FreeImage)
- 
--find_package(Metis REQUIRED)
-+find_package(metis CONFIG REQUIRED)
-+set(METIS_LIBRARIES metis)
- 
- find_package(Glog REQUIRED)
- 
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (JonLiu1993) [inline ports/colmap/fix-dependencies.patch:1]: @ahojnnes, please restore the patch format.

**Comment by JonLiu1993** [comment]
Please fix the conflicting files.

File: ports/colmap/fix-dependencies.patch (hunk 6 above)
Comment (ahojnnes) [reply ports/colmap/fix-dependencies.patch:1]: Apologies, but I don't understand what you mean by this comment? Which patch format?

Comment (ahojnnes) [reply ports/colmap/fix-dependencies.patch:1]: The underlying CMakeLists.txt changed and this patch requires an update.

**Comment by JonLiu1993** [comment]
@ahojnnes, When I tested the features by command "./vcpkg install colmap[*]:x64-windows" I get this error.
Please take a look:
````
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "colmap" of type SHARED_LIBRARY
    depends on "colmap_cuda" (weak)
  "colmap_cuda" of type SHARED_LIBRARY
    depends on "colmap" (weak)
At least one of these targets is not a STATIC_LIBRARY.  Cyclic dependencies are allowed only among static libraries.
CMake Generate step failed.  Build files cannot be regenerated correctly.
````

**Comment by ahojnnes** [comment]
@JonLiu1993 I believe this issue is fixed now.

**Comment by ahojnnes** [comment]
It required a fix in the upstream repo.

**Review by JonLiu1993 — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**